iceoryx_hoofs/include/iceoryx_hoofs/log/logging.hpp

```diff
@@ -13,97 +13,46 @@
 
-#include "iceoryx_hoofs/log/logcommon.hpp"
-#include "iceoryx_hoofs/log/logger.hpp"
 #include "iceoryx_hoofs/log/logstream.hpp"
-
-#include <chrono>
-#include <string>
-
 namespace iox
 {
 namespace log
 {
-Logger& createLogger(const std::string& ctxId,
-                     const std::string& ctxDescription,
-                     const LogLevel appDefLogLevel = LogLevel::kWarn) noexcept;
-
-inline constexpr LogHex8 HexFormat(uint8_t value) noexcept
-{
-    return LogHex8(value);
-}
-inline constexpr LogHex8 HexFormat(int8_t value) noexcept
-{
-    return LogHex8(static_cast<uint8_t>(value));
-}
-inline constexpr LogHex16 HexFormat(uint16_t value) noexcept
-{
-    return LogHex16(value);
-}
-inline constexpr LogHex16 HexFormat(int16_t value) noexcept
-{
-    return LogHex16(static_cast<uint16_t>(value));
-}
-inline constexpr LogHex32 HexFormat(uint32_t value) noexcept
-{
-    return LogHex32(value);
-}
-inline constexpr LogHex32 HexFormat(int32_t value) noexcept
-{
-    return LogHex32(static_cast<uint32_t>(value));
-}
-inline constexpr LogHex64 HexFormat(uint64_t value) noexcept
-{
-    return LogHex64(value);
-}
-inline constexpr LogHex64 HexFormat(int64_t value) noexcept
-{
-    return LogHex64(static_cast<uint64_t>(value));
-}
-
-inline constexpr LogBin8 BinFormat(uint8_t value) noexcept
-{
-    return LogBin8(value);
-}
-inline constexpr LogBin8 BinFormat(int8_t value) noexcept
-{
-    return LogBin8(static_cast<uint8_t>(value));
-}
-inline constexpr LogBin16 BinFormat(uint16_t value) noexcept
-{
-    return LogBin16(value);
-}
-inline constexpr LogBin16 BinFormat(int16_t value) noexcept
-{
-    return LogBin16(static_cast<uint16_t>(value));
-}
-inline constexpr LogBin32 BinFormat(uint32_t value) noexcept
-{
-    return LogBin32(value);
-}
-inline constexpr LogBin32 BinFormat(int32_t value) noexcept
-{
-    return LogBin32(static_cast<uint32_t>(value));
-}
-inline constexpr LogBin64 BinFormat(uint64_t value) noexcept
-{
-    return LogBin64(value);
-}
-inline constexpr LogBin64 BinFormat(int64_t value) noexcept
-{
-    return LogBin64(static_cast<uint64_t>(value));
-}
-
-template <typename T, typename std::enable_if<!std::is_pointer<T>::value, std::nullptr_t>::type = nullptr>
-inline constexpr LogRawBuffer RawBuffer(const T& value) noexcept
-{
-    // LogRawBuffer is used with the streaming operator which converts the data into a string,
-    // therefore we shouldn't get lifetime issues
-    return LogRawBuffer{reinterpret_cast<const uint8_t*>(&value), sizeof(T)};
-}
+/// @brief Only for internal usage
+// NOLINTJUSTIFICATION cannot be realized with templates or constexpr functions due to the the source location intrinsic
+// NOLINTBEGIN(cppcoreguidelines-macro-usage)
+#define IOX_LOG_INTERNAL(file, line, function, level)                                                                  \
+    if ((level) <= iox::log::Logger::minimalLogLevel()                                                                 \
```

Comment:
@MatthiasKillat it is for cases where the filtering is not supposed to happen on the client side but on the logger side, e.g. the logger want's to forward all messages independent of what the user has set as log level.

Reply:
This will come at the prize of efficiency. If we need more efficiency it can be a compile time choice. constexpr and restructuring some ifs to allow the compiler to remove branches at compile time could be enough (not sure).

Reply:
@MatthiasKillat exactly, it is a compile time option which is set to `false` by default